cmake/FlatccGenerateSources.cmake

```diff
@@ -0,0 +1,340 @@
+        if(IS_ABSOLUTE "${schema_file}")
+            set(absolute_def_file "${schema_file}")
+        else()
+            set(absolute_def_file "${CMAKE_CURRENT_SOURCE_DIR}/${schema_file}")
+        endif()
+        list(APPEND ABSOLUTE_SCHEMA_FILES "${absolute_def_file}")
+
+        get_filename_component(absolute_def_directory "${absolute_def_file}" DIRECTORY)
+        list(APPEND absolute_flatcc_paths "${absolute_def_directory}")
+    endforeach()
+
+    foreach(path ${FLATCC_PATHS})
+        # TODO: file(REAL_PATH) if cmake_minimum_required_version >= 3.19
+        if(NOT IS_ABSOLUTE "${path}")
+            set(path "${CMAKE_CURRENT_SOURCE_DIR}/${path}")
+        endif()
+        list(APPEND absolute_flatcc_paths "${path}")
+        list(APPEND FLATCC_ARGS -I "${path}")
+    endforeach()
+    list(REMOVE_DUPLICATES absolute_flatcc_paths)
+
+    # if no option is passed, the default READER is used
+    set(default_option_reader ON)
+    foreach(output_option ${output_options})
+        if(output_option)
+            set(default_option_reader OFF)
+        endif()
+    endforeach()
+    if(default_option_reader)
+        set(FLATCC_READER ON)
+    endif()
+
+    # Add flatcc options
+
+    if(FLATCC_PREFIX)
+        list(APPEND FLATCC_ARGS "--prefix=${FLATCC_PREFIX}")
+    endif()
+
+    # Handle reader option first as other option encompass reader
+    if (FLATCC_READER)
+        list(APPEND FLATCC_ARGS --reader)
+    endif()
+    if (FLATCC_COMMON_READER)
+        list(APPEND FLATCC_ARGS --common)
+    endif()
+    if (FLATCC_COMMON_BUILDER)
+        list(APPEND FLATCC_ARGS --common)
+    endif()
+    if (FLATCC_BUILDER)
+        list(APPEND FLATCC_ARGS --builder)
+        # Builder also generates reader
+        set(FLATCC_READER ON)
+    endif()
+    if (FLATCC_VERIFIER)
+        list(APPEND FLATCC_ARGS --verifier)
+        # verifier also generates reader
+        set(FLATCC_READER ON)
+    endif()
+    if (FLATCC_JSON_PARSER)
+        list(APPEND FLATCC_ARGS --json-parser)
+    endif()
+    if (FLATCC_JSON_PRINTER)
+        list(APPEND FLATCC_ARGS --json-printer)
+    endif()
+    if (FLATCC_BINARY_SCHEMA)
+        list(APPEND FLATCC_ARGS --schema)
+    endif()
+    if (FLATCC_RECURSIVE)
+        list(APPEND FLATCC_ARGS --recursive)
+    endif()
+
+    # handle 'all', 'common', 'json' last as they encompass other options
+    if (FLATCC_COMMON)
+        list(APPEND FLATCC_ARGS --common)
+        set(FLATCC_COMMON_READER ON)
+        set(FLATCC_COMMON_BUILDER ON)
+    endif()
+    if (FLATCC_JSON)
+        list(APPEND FLATCC_ARGS --json)
+        set(FLATCC_JSON_PARSER ON)
+        set(FLATCC_JSON_PRINTER ON)
+    endif()
+    if (FLATCC_ALL)
+        list(APPEND FLATCC_ARGS -a)
+        set(FLATCC_COMMON_BUILDER ON)
+        set(FLATCC_COMMON_READER ON)
+        set(FLATCC_BUILDER ON)
+        set(FLATCC_VERIFIER ON)
+        set(FLATCC_READER ON)
+        set(FLATCC_RECURSIVE ON)
+    endif()
```

Comment:
`ALL` implies common (c), builder (w), verifier (v), reader and recursive (r).
Reader is built always.